### PR TITLE
Providers cleanup

### PR DESCRIPTION
**Pull request by miigotu**
Can't find the error here. I just want to save this in case I can find it but the entire page needs rewritten in the new UI anyway, and the providers need loaded with stevedore, as does notifiers and other items. Possibly something other than stevedore but I am going to work on that and the new UI instead of this for now.

## Summary by CodeRabbit

- **New Feature**
  - Enhanced provider configuration for NZB and torrent searches, offering more customization options.
- **Improvement**
  - Streamlined provider migration and updating process, improving user experience.
  - Updated show list display on the home page for better organization and readability.
- **Refactor**
  - Improved code readability and maintainability across various modules.
  - Updated variable names and function calls for better code clarity and consistency.
- **Security**
  - Enhanced SQL query security in the `History` class to prevent SQL injection.
- **Bug Fix**
  - Fixed potential issues with provider URL handling in the `GenericProvider` class.
- **Documentation**
  - Added Qodana analysis configuration for better code quality checks.

### CODE REVIEW

**Comment by miigotu** [comment]
@coderabbitai review

File: sickchill/providers/GenericProvider.py [hunk 1]
```diff
@@ -606,31 +606,35 @@
...
+        custom_url_valid = True
+        if self.has_option("custom_url") and self.has_option(self, "url") and self.has_option(self, "urls"):
+            has_custom_url = bool(self.custom_url)
+            custom_url_valid = self.valid_url(self.custom_url)
+            if has_custom_url and custom_url_valid:
+                self.__store_original_urls()
+
+                for url in self.__original_urls:
+                    # update each url in our urls to have the correct proxy url
+                    self.urls[url] = urljoin(self.custom_url, self.__original_urls[url].split(self.__original_url)[1])
```
[the remark this answers is not in the document]
Comment (miigotu) [reply sickchill/providers/GenericProvider.py:636]: thats the intention...

File: sickchill/gui/slick/views/config_providers.mako [hunk 2]
```diff
@@ -228,107 +200,92 @@
...
-                                            <input type="checkbox" name="${curNewznabProvider.get_id("_search_fallback")}"
-                                                   id="${curNewznabProvider.get_id("_search_fallback")}" ${checked(curNewznabProvider.search_fallback_enabled)}/>
-                                            <label for="${curNewznabProvider.get_id("_search_fallback")}">${_('when searching for a complete season depending on search mode you may return no results, this helps by restarting the search using the opposite search mode.')}</label>
+                                            <input type="checkbox" name="${provider.get_id("_search_fallback")}" id="${provider.get_id("_search_fallback")}" ${checked(provider.search_fallback_enabled)}/>
+                                            <label for="${provider.get_id("_search_fallback")}">
+                                                ${_('when searching for a complete season depending on search mode you may return no results, this helps by restarting the search using the opposite search mode.')}
+                                            ss</label>
```
[the remark this answers is not in the document]
Comment (miigotu) [reply sickchill/gui/slick/views/config_providers.mako:226]: @BKSteve I missed this lol, looks like a Ctrl+sssss 🤷‍♂️🤣

[the remark this answers is not in the document]
Comment (BKSteve) [reply sickchill/gui/slick/views/config_providers.mako:226]: Halloween prep? Or snakes on a plane? :snake:  :snake: :airplane: 
The rabbit showing its usefulness!

[the remark this answers is not in the document]
Comment (BKSteve) [reply sickchill/gui/slick/views/config_providers.mako:226]: But no easy accept box, have to do the changes suggested by ourselves. 
As you merged and this issue is still there. patch-1 raised

**Pull request merged**